presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java

```diff
@@ -146,14 +146,49 @@ public Page getOutput()
             return null;
         }
 
-        String plan = textDistributedPlan(queryInfo.getOutputStage().get().getSubStages(), metadata, operatorContext.getSession());
+        String plan = textDistributedPlan(ImmutableList.of(queryInfo.getOutputStage().get()), metadata, operatorContext.getSession());
+        String printPlan = removeExplainAnalyzeFragment(plan);
         BlockBuilder builder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), 1);
-        VARCHAR.writeString(builder, plan);
+        VARCHAR.writeString(builder, printPlan);
 
         outputConsumed = true;
         return new Page(builder.build());
     }
 
+    private String removeExplainAnalyzeFragment(String plan)
```

Comment:
This is the wrong approach - I would create another QueryInfo instead. This is too brittle and would break anytime we change the way the plan is printed.